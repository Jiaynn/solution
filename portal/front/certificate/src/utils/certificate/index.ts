/*
 * @file component certificate utils
 * @author Yao Jingtian <yncst233@gmail.com>
 */

import { sortBy, last, without, uniq, values } from 'lodash'

import { OrderType } from '../../apis/ssl'
import {
  sslMap, canBeClosedStatuses, SSLDomainType, CertType,
  OrderStatus, ProductShortName, SslBrand, isDVSSLType
} from '../../constants/ssl'
import { isOEM } from '../../constants/env'
import { isStatusDone, isStatusPending } from './status'

interface ICertBrandInfo {
  brand: SslBrand
  certType: CertType
}

// 根据 productShortName 获取证书品牌、类型
export function shortNameToInfo(productShortName?: ProductShortName): ICertBrandInfo {
  let brand: SslBrand
  let certType: CertType
  Object.keys(sslMap).some((sslBrand: SslBrand) => Object.keys(sslMap[sslBrand]).some((sslType: CertType) => {
    if (sslMap[sslBrand][sslType]!.shortName === productShortName) {
      brand = sslBrand
      certType = sslType
      return true
    }
    if (productShortName && sslMap[sslBrand][sslType]!.domainShortNameMap) {
      const names = values(sslMap[sslBrand][sslType]!.domainShortNameMap)
      if (names.indexOf(productShortName) > -1) {
        brand = sslBrand
        certType = sslType
        return true
      }
    }
    return false
  }))
  return {
    brand: brand!,
    certType: certType!
  }
}

export function getLatestDnsNames(replaceOrders: any[], dnsNames?: string[]): string[] {
  if (!replaceOrders || replaceOrders.length === 0) {
    return dnsNames || []
  }
  const list = sortBy(replaceOrders, 'create_time')
    .filter(order => order.state === 1)
  if (list.length > 0) {
    return last(list).dnsnames.split(',')
  }
  return dnsNames || []
}

export function hasRepeatItem(arr: any[], compareArr?: any[]) {
  if (arr && compareArr) {
    return without(arr.slice(), ...compareArr.slice()).length !== arr.length
  }
  return uniq(arr.slice()).length !== arr.length
}

/**
 * 从后端接口字段 renewable 来判断
 * 历史前端逻辑：已签发/已重颁发 过期前 30 天的非重颁发订单，显示【续费】(若 notAfter <= 0，则根据 providerRenewable 字段)
 */
export function canRenew(renewable: boolean) {
  return isOEM ? false : renewable
}

interface ICanAddDomainParams {
  notAfter: number,
  state: number,
  productType: SSLDomainType,
  orderType: OrderType
}

// 已签发/已重颁发 非重颁发 未过期的多域名、多域名泛域名订单，显示【添加域名】
export function canAddDomain(params: ICanAddDomainParams) {
  const { notAfter, state, productType, orderType } = params
  const now = Date.now()
  if (!state || !isStatusDone(state)) {
    return false
  }
  if ([SSLDomainType.Multiple, SSLDomainType.MultipleWildcard].indexOf(productType) === -1) {
    return false
  }
  if (orderType === OrderType.Replace) {
    return false
  }
  if (!notAfter) {
    return false
  }
  return (
    notAfter <= 0
    || now < notAfter * 1000
  )
}

interface ICanUploadConfirmationParams {
  state: number,
  productShortName: ProductShortName,
  uploadConfirmLetter: boolean
}

// 非 DV 且待确认状态才显示【上传确认函】
export function canUploadConfirmation(params: ICanUploadConfirmationParams) {
  const { state, productShortName, uploadConfirmLetter } = params
  if (!state || !isStatusPending(state)) {
    return false
  }

  const sslInfo = shortNameToInfo(productShortName)

  return !uploadConfirmLetter && !isDVSSLType(sslInfo.certType)
}

export function canClose(state: number): boolean {
  return canBeClosedStatuses.indexOf(state) !== -1
}

// 过期 7 天前的证书都能够部署
export function canCertDeploy(notAfter: number): boolean {
  return !isOEM && ((notAfter - 7 * 24 * 3600) > (Date.now() / 1000))
}

// 判断证书类型是否为私有证书
export function isPrivateCert(productShortName: ProductShortName) {
  return !productShortName
}

// 已经完成的订单，并且满足证书部署条件的订单能够部署
export function canOrderDeploy(notAfter: number, state: OrderStatus): boolean {
  return isStatusDone(state) && canCertDeploy(notAfter)
}

// 2 年的续期证书需要补充文案
export function getCertYearTipsByYear(year: number): string {
  return year === 2 ? '续期证书' : ''
}

// 自动续期的证书、订单需要补充文案
export function getCertYearTipsByAutoRenew(autoRenew: boolean): string {
  return autoRenew ? '续期证书' : ''
}
