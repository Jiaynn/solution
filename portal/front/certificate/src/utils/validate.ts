/*
 * @file formstate validate
 * @author zhu hao <zhuhao@qiniu.com>
 */

import { standardDomainRegx, wildcardDomainRegx } from '../constants/domain'

export function validate(criteria: boolean, invalidMsg: string) {
  return criteria ? invalidMsg : null
}

export function notEmpty(value: unknown, name = '') {
  return validate(value == null || !value, `${name}不可以为空`)
}

export function inputValid(value: string) {
  return validate(/[`~～!！@#$%^&*_+<>?:"“”{},，。./;；'‘’[\]]/.test(value), '不得含有非法字符')
}

export function mustContainChinese(value: string, name = '') {
  return validate(!/[\u4e00-\u9fa5]/.test(value), `${name}应包含中文字符`)
}

export function postCodeValid(value: string) {
  return validate(!/^\d{5,6}(?!\d)$/.test(value), '邮编格式不正确')
}

export function areaCodeValid(value: string) {
  return validate(!/^((0\d{2,3}))$/.test(value), '区号格式不正确')
}

export function telephoneValid(value: string) {
  return validate(!/^(\d{7,8})(-(\d{3,}))?$/.test(value), '座机号格式不正确')
}

export function mobilephoneValid(value: string) {
  return validate(!/^\d{11}$/.test(value), '手机号格式不正确')
}

export function emailValid(value: string) {
  return validate(!/^\w+([-+.]\w+)*@\w+([-.]\w+)*\.\w+([-.]\w+)*$/.test(value), '邮箱格式不正确')
}

export function wildcardDomainValid(value: string) {
  return validate(!wildcardDomainRegx.test(value), '泛域名格式不正确')
}

export function standardDomainValid(value: string) {
  return validate(!standardDomainRegx.test(value), '标准域名格式不正确')
}

export function onlySupportChineseAndCharacter(value: string) {
  return validate(!/^[a-zA-Z0-9_\s\-*.()\u4e00-\u9fa5]+$/.test(value), '仅支持中英文字符、数字、空格、_、-、*、.、()')
}
