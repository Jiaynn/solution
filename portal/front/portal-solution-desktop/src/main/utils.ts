import { join } from 'path'
import { BrowserWindow, dialog, session, shell } from 'electron'
import cmd from 'node-cmd'

import { pageUrl } from './config'
import { ElectronBridgeApi } from '../preload'

import icon from '../../resources/logo.png?asset'

export const createWindow = (): BrowserWindow => {
  // Create the browser window.
  const minWidth = 1024
  const minHeight = 768
  const mainWindow = new BrowserWindow({
    width: minWidth,
    height: minHeight,
    minWidth,
    minHeight,
    show: false,
    autoHideMenuBar: true,
    titleBarStyle: 'hidden',
    ...(process.platform === 'linux' ? { icon } : {}),
    webPreferences: {
      preload: join(__dirname, '../preload/index.js'),
      sandbox: false,
      nodeIntegration: true
    }
  })

  mainWindow.center()

  mainWindow.on('ready-to-show', () => {
    mainWindow.show()
  })

  mainWindow.webContents.setWindowOpenHandler((details) => {
    shell.openExternal(details.url)
    return { action: 'deny' }
  })

  // HMR for renderer base on electron-vite cli.
  // Load the remote URL for development or the local html file for production.
  mainWindow.loadURL(pageUrl)

  return mainWindow
}

export const persistentCookie = (): void => {
  session.defaultSession.cookies.on('changed', (_, cookie) => {
    if (cookie.session && cookie.name === 'PORTAL_SESSION') {
      session.defaultSession.cookies.set({
        url: pageUrl,
        name: cookie.name,
        value: cookie.value,
        domain: cookie.domain,
        path: cookie.path,
        secure: cookie.secure,
        httpOnly: cookie.httpOnly,
        expirationDate: Math.floor(Date.now() / 1000) + 60 * 60 * 24 * 2,
        sameSite: cookie.sameSite
      })
    }
  })
}

type Platform = Parameters<ElectronBridgeApi['openEditor']>[0]['platform']
export const callEditor = (platform: Platform, dest: string): void => {
  if (process.platform === 'darwin') {
    // 运行在 macOS 上
    if (platform === 'Android') {
      cmd.run(`open -a /Applications/Android\\ Studio.app ${dest}`)
      return
    }
    if (platform === 'iOS') {
      cmd.run(`open -a /Applications/Xcode.app ${dest}`)
      return
    }
  }
  if (process.platform === 'win32') {
    // 运行在 Windows 上
    // ...
    return
  }
  if (process.platform === 'linux') {
    // 运行在 Linux 上
    // ...
  }
  dialog.showMessageBox({ message: '打开失败，请检查编辑器是否安装或项目是否存在' })
}
