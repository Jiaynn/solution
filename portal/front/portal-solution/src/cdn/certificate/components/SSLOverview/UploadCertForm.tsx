/*
 * @file component UploadCertForm in SSLOverview
 * @author Yao Jingtian <yncst233@gmail.com>
 */

import React from 'react'
import { observable, action, computed, makeObservable } from 'mobx'
import { observer } from 'mobx-react'

import Button from 'react-icecream/lib/button'
import Form from 'react-icecream/lib/form'
import Input from 'react-icecream/lib/input'
import Radio, { RadioChangeEvent } from 'react-icecream/lib/radio'
import Col from 'react-icecream/lib/col'
import Alert from 'react-icecream/lib/alert'

import { useInjection } from 'qn-fe-core/di'
import { ToasterStore } from 'portal-base/common/toaster'
import { Loadings } from 'portal-base/common/loading'

import SslApis from '../../apis/ssl'
import { SSLFormItem } from '../SSLFormItem'

import './style.less'

const FormItem = Form.Item
const createForm = Form.create
const RadioGroup = Radio.Group

interface IUploadCertFormProps {
  onUploaded: () => void,
  onCancel: () => void
}
interface IAddonFormProps {
  form?: any
}

export type UploadCertFormProps = IUploadCertFormProps & IAddonFormProps

type UploadCertFormInnerProps = UploadCertFormProps & {
  sslApis: SslApis
  toasterStore: ToasterStore
}

type UploadValue = {
  name: string
  ca: string
  pri: string
}

@(createForm as any)()
@observer
export class UploadCertForm extends React.Component<UploadCertFormInnerProps> {
  constructor(props: UploadCertFormInnerProps) {
    super(props)
    makeObservable(this)
    ToasterStore.bindTo(this, props.toasterStore)
    this.handleCancel = this.handleCancel.bind(this)
    this.updateStep = this.updateStep.bind(this)
    this.updateCA = this.updateCA.bind(this)
    this.updateUploadData = this.updateUploadData.bind(this)
    this.initUploadCert = this.initUploadCert.bind(this)
    this.handleRadioChange = this.handleRadioChange.bind(this)
    this.checkIfShouldConfirm = this.checkIfShouldConfirm.bind(this)
    this.save = this.save.bind(this)
    this.handleUploadBtnClick = this.handleUploadBtnClick.bind(this)
    this.handleConfirmBtnClick = this.handleConfirmBtnClick.bind(this)
  }

  loadings = new Loadings('upload')
  @observable caid = 0
  @observable uploadData: UploadValue = {
    name: '',
    ca: '',
    pri: ''
  }
  @observable userCA = ''
  @observable serverCA = ''
  @observable uploadStep = 1

  @computed get isUploading() {
    return this.loadings.isLoading('upload')
  }

  @action handleCancel() {
    this.props.onCancel()
    this.initUploadCert()
  }

  @action handleRadioChange(e: RadioChangeEvent) {
    this.caid = e.target.value
  }

  @action updateCA(serverCa: string, userCa: string) {
    if (serverCa) {
      this.serverCA = serverCa
    }
    if (userCa) {
      this.userCA = userCa
    }
  }

  @action updateStep(step: number) {
    this.uploadStep = step
  }

  @action updateUploadData(values: UploadValue) {
    this.uploadData = {
      name: values.name,
      ca: values.ca,
      pri: values.pri
    }
  }

  @action initUploadCert() {
    this.updateStep(1)
    this.props.form.resetFields()
  }

  @action save() {
    return this.props.sslApis.saveSSLCert(this.uploadData).then(() => {
      this.initUploadCert()
      this.props.onUploaded()
    })
  }

  checkIfShouldConfirm(values: UploadValue) {
    return this.props.sslApis.fetchCert(values.ca).then(res => {
      if (values.ca.split('-----END CERTIFICATE-----').length <= 2) {
        // 一段CA，用拉取的补全证书提交
        values.ca = res.ca
        this.updateUploadData(values)
        return false
      }
      // 两段及以上，比对
      const serverCa = res.ca.replace(/\r/g, '').replace(/\n/g, '').replace(/\s/g, '')
      const userCa = values.ca.replace(/\r/g, '').replace(/\n/g, '').replace(/\s/g, '')
      if (userCa === serverCa) {
        return false
      }
      // 需要确认
      this.updateCA(res.ca, values.ca)
      return true
    })
  }

  @ToasterStore.handle()
  checkAndSubmit(values: UploadValue) {
    // 检查是否需要确认
    const result = this.checkIfShouldConfirm(values)
      .then(shouldConfirm => {
        if (shouldConfirm) {
          this.updateStep(2)
          return
        }
        return this.save().then(
          () => this.props.toasterStore.success('证书上传成功！')
        )
      }).catch(error => {
        this.initUploadCert()
        return Promise.reject(error)
      })

    return this.loadings.promise('upload', result)
  }

  handleUploadBtnClick() {
    return this.props.form.validateFields((errors: unknown, values: UploadValue) => {
      if (errors) {
        return
      }
      Object.keys(values).forEach((key: keyof UploadValue) => {
        values[key] = values[key].trim()
      })
      this.updateUploadData(values)
      this.checkAndSubmit(values)
    })
  }

  @ToasterStore.handle('证书上传成功！')
  handleConfirmBtnClick() {
    this.updateUploadData({
      ca: this.caid === 0 ? this.userCA : this.serverCA,
      pri: this.uploadData.pri.trim(),
      name: this.uploadData.name
    })
    const result = this.save().catch(error => {
      this.initUploadCert()
      return Promise.reject(error)
    })
    return this.loadings.promise('upload', result)
  }

  render() {
    const lightboxLeftAlignFormItemLayout = {
      labelCol: { span: 6 },
      wrapperCol: { span: 18 }
    }
    const { getFieldDecorator } = this.props.form

    const extraProps: any = { rows: 10 }

    return (
      <Form layout="horizontal">
        <div className={this.uploadStep === 2 ? 'hidden' : ''}>
          <Alert
            className="upload-warn"
            message={<span>请上传有效期不小于 30 天的证书。目前只支持 PEM 格式，其它格式请前往 <a className="link" href="https://myssl.com/cert_convert.html" target="_blank" rel="noopener">该站点</a> 转换。</span>}
            type="warning"
            showIcon
            closable={false}
          />
          <SSLFormItem {...{
            label: '证书备注名',
            required: true,
            layout: lightboxLeftAlignFormItemLayout,
            fieldName: 'name',
            getFieldDecorator,
            rules: [
              { required: true, message: '必填' },
              { pattern: /^[a-zA-Z0-9_\s\-*.()\u4e00-\u9fa5]+$/, message: '仅支持中英文字符、数字、空格、_、-、*、.、()' },
              { whitespace: true, message: '不可以为空' }
            ],
            component: <Input />
          }} />
          <SSLFormItem {...{
            label: '证书内容 ( PEM格式 )',
            required: true,
            layout: lightboxLeftAlignFormItemLayout,
            fieldName: 'ca',
            getFieldDecorator,
            rules: [
              { required: true, message: '必填' },
              { whitespace: true, message: '不可以为空' }
            ],
            component: <Input.TextArea
              placeholder="请将证书内容(包含证书链)复制粘贴到此处"
              {...extraProps}
            />
          }} />
          <SSLFormItem {...{
            label: '证书私钥 ( PEM格式 )',
            required: true,
            layout: lightboxLeftAlignFormItemLayout,
            fieldName: 'pri',
            getFieldDecorator,
            rules: [
              { required: true, message: '必填' },
              { whitespace: true, message: '不可以为空' }
            ],
            component: <Input.TextArea
              placeholder="请将证书私钥复制粘贴到此处"
              {...extraProps}
            />
          }} />
        </div>
        {
          this.uploadStep === 2
          ? <FormItem required>
            <Alert
              message="系统检测到您上传的证书内容与系统自动补全的不一致，请选择一个版本并上传。"
              type="warning"
              showIcon
              closable={false}
            />
            <a className="help-btn" href="https://developer.qiniu.com/ssl/kb/3703/the-certificate-chain-is-what" target="_blank" rel="noopener">帮助</a>
            <RadioGroup onChange={this.handleRadioChange} value={this.caid}>
              <Radio key="a" value={0} style={{ width: '50%' }}>原自有证书内容</Radio>
              <Radio key="b" value={1} style={{ width: '50%' }}>系统补全证书内容</Radio>
            </RadioGroup>
            <Col span={12}>
              <pre className="cert-ca">{this.userCA}</pre>
            </Col>
            <Col span={12}>
              <pre className="cert-ca">{this.serverCA}</pre>
            </Col>
          </FormItem>
          : null
        }
        <hr className="split-line sub-split-line" />
        <div className="footer-center">
          <Button key="back" type="ghost" onClick={this.handleCancel}>取消</Button>
          <Button key="submit" type="primary" disabled={this.isUploading} onClick={this.uploadStep === 1 ? this.handleUploadBtnClick : this.handleConfirmBtnClick}>确定上传</Button>
        </div>
      </Form>
    )
  }
}

export default observer(function _UploadCertForm(props: UploadCertFormProps) {
  const sslApis = useInjection(SslApis)
  const toasterStore = useInjection(ToasterStore)
  return <UploadCertForm {...props} sslApis={sslApis} toasterStore={toasterStore} />
})
