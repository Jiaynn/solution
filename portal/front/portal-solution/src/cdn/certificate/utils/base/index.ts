/*
 * @file base utils
 * @author nighca <nighca@live.cn>
 */

import moment from 'moment'

import { RouterStore } from 'portal-base/common/router'

export interface IPayOrderOptions {
  orderId: string       // fusion 订单号
  tradeOrderId: string  // BO 订单号
}

export function payForOrder({ tradeOrderId }: IPayOrderOptions, routerStore: RouterStore) {
  routerStore.push(`/financial/verify-order/${tradeOrderId}`)
}

export function humanizeTime(time: number, format?: string) {
  return format
    ? moment(time * 1000).format(format)
    : moment(time * 1000).format('YYYY-MM-DD HH:mm:ss')
}
