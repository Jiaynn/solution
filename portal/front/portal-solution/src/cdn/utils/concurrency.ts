/**
 * @file concurrency limit request
 * @source https://gist.github.com/nighca/6562d098ac01814b6e1c1718b16d4abc
 */

export default function concurrency<T>(process: (_: any) => Promise<T>, limit = -1) {
  return function batchProcess(tasks: unknown[] = []): Promise<T[]> {
    const results: T[] = []
    let finished = 0
    let current = 0
    let rejected = false

    function tryProcess(resolve: (...args: any) => any, reject: (...args: any) => any) {
      if (rejected) return
      if (finished >= tasks.length) return resolve(results)
      if (current >= tasks.length) return

      const index = current++
      process(tasks[index]).then(
        result => {
          results[index] = result
          finished++
          tryProcess(resolve, reject)
        },
        err => {
          reject(err)
          rejected = true
        }
      )
    }

    return new Promise((resolve, reject) => {
      const realLimit = limit > 0 ? limit : tasks.length
      for (let i = 0; i < realLimit; i++) {
        tryProcess(resolve, reject)
      }
    })
  }
}
