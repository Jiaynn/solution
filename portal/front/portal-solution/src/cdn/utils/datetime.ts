import moment from 'moment'

export function isLastDayOfMonth() {
  return moment().endOf('month').format('MM-DD') === moment().format('MM-DD')
}
