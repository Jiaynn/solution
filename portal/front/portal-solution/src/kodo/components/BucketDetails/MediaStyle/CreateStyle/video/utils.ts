export enum SourceFormat {
  Mp4 = 'mp4',
  Flv = 'flv',
  Mov = 'mov',
}

export enum OutputFormat {
  Jpg = 'jpg',
  Png = 'png',
  Gif = 'gif'
}

export enum CoverType {
  Static = 'static',
  Dynamic = 'dynamic'
}

export const coverTypeNameMap = {
  [CoverType.Static]: '静态封面',
  [CoverType.Dynamic]: '动态封面'
}

export const coverTypes = [CoverType.Static] // [CoverType.Static, CoverType.Dynamic]
export const sourceFormatList = [SourceFormat.Mp4, SourceFormat.Flv, SourceFormat.Mov]
export const allowPreviewMimeTypeList = [
  'video/mp4',
  'video/x-flv',
  'video/quicktime'
]

export const allowPickAccept = {
  name: '视频',
  maxSize: 500 * 1024 * 1024,
  mimeTypes: allowPreviewMimeTypeList
}

export const outputFormatList = [OutputFormat.Jpg, OutputFormat.Png, OutputFormat.Gif]
export const staticCoverSuffixes = [OutputFormat.Jpg, OutputFormat.Png]
export const dynamicCoverSuffixes = [OutputFormat.Gif]

export enum CoverScaleType {
  None = 'none',
  // 限定宽, 高同比例缩放
  ScaleWidth = 'scaleWidth',
  // 限定高, 宽同比例缩放
  ScaleHeight = 'scaleHeight',
  // 限定宽高同比例缩放
  ScaleWidthHeight = 'scaleWidthHeight',
}

export const coverScaleTypes = [
  CoverScaleType.None,
  CoverScaleType.ScaleWidth,
  CoverScaleType.ScaleHeight,
  CoverScaleType.ScaleWidthHeight
]

export const coverScaleTypeNameMap = {
  [CoverScaleType.None]: '不缩放',
  [CoverScaleType.ScaleWidth]: '限定宽、高同比例缩放',
  [CoverScaleType.ScaleHeight]: '限定高、宽同比例缩放',
  [CoverScaleType.ScaleWidthHeight]: '限定宽高同比例缩放'
}

export enum CoverAutoScaleType {
  None = 0,
  One = 1,
  Two = 2
}

export const coverAutoScaleTypeNameMap = {
  [CoverAutoScaleType.None]: '不启用',
  [CoverAutoScaleType.One]: 1,
  [CoverAutoScaleType.Two]: 2
}

export const coverAutoScaleTypes = [CoverAutoScaleType.None, CoverAutoScaleType.One, CoverAutoScaleType.Two]

export function getCoverScaleType(w: number | null, h: number | null) {
  if (w == null && h == null) {
    return CoverScaleType.None
  }
  if (w != null && h == null) {
    return CoverScaleType.ScaleWidth
  }

  if (w == null && h != null) {
    return CoverScaleType.ScaleHeight
  }

  return CoverScaleType.ScaleWidthHeight
}

export function isValidSourceFormat(format: string): format is SourceFormat {
  return sourceFormatList.includes(format as any)
}

export function encodeKodoURI(bucketName: string, fileKey: string) {
  return `kodo://${bucketName}/${fileKey}`
}

export function decodeKodoURI(URI: string): [bucketName: string, fileKey: string] {
  const execArray = /kodo:\/\/([^/]+)\/(.*)/.exec(URI)
  if (execArray) {
    return execArray.slice(1, 3) as [string, string]
  }
  // 不符合规则时，直接把输入当作 fileKey 返回
  return ['', URI]
}
