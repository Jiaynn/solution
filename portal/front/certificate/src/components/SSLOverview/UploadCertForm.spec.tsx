/**
 * @file test cases for component UploadCertForm
 * @author [Yao Jingtian] [yncst233@gmail.com]
 */
import React from 'react'

import { RendererUtils as Renderer } from '../../utils/test'
import UploadCertForm from './UploadCertForm'

const onUploaded = () => { console.log('uploaded') }
const onCancel = () => { console.log('canceled') }

it('renders correctly', () => {
  const renderer = new Renderer()
  const tree = renderer.create(
    <UploadCertForm onUploaded={onUploaded} onCancel={onCancel} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
