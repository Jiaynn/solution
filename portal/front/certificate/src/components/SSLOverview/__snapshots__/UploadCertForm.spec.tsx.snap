// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="ant-form ant-form-horizontal"
  onSubmit={[Function]}
>
  <div
    className=""
  >
    <div
      className="ant-alert ant-alert-warning upload-warn"
      data-show={true}
    >
      <i
        aria-label="icon: warning"
        className="anticon anticon-warning ant-alert-icon"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="warning"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M464 720a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm16-304v184c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V416c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8zm475.7 440l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zm-783.5-27.9L512 239.9l339.8 588.2H172.2z"
          />
        </svg>
      </i>
      <span
        className="ant-alert-message"
      >
        <span>
          请上传有效期不小于 30 天的证书。目前只支持 PEM 格式，其它格式请前往 
          <a
            className="link"
            href="https://myssl.com/cert_convert.html"
            rel="noopener"
            target="_blank"
          >
            该站点
          </a>
           转换。
        </span>
      </span>
      <span
        className="ant-alert-description"
      />
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-6 ant-form-item-label"
      >
        <label
          className="ant-form-item-required"
          htmlFor="name"
          onClick={[Function]}
          title="证书备注名"
        >
          证书备注名
        </label>
      </div>
      <div
        className="ant-col ant-col-18 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <input
              className="ant-input"
              data-__field={
                Object {
                  "name": "name",
                }
              }
              data-__meta={
                Object {
                  "initialValue": undefined,
                  "name": "name",
                  "originalProps": Object {
                    "disabled": false,
                    "type": "text",
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "必填",
                      "required": true,
                    },
                    Object {
                      "message": "仅支持中英文字符、数字、空格、_、-、*、.、()",
                      "pattern": /\\^\\[a-zA-Z0-9_\\\\s\\\\-\\*\\.\\(\\)\\\\u4e00-\\\\u9fa5\\]\\+\\$/,
                    },
                    Object {
                      "message": "不可以为空",
                      "whitespace": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "必填",
                          "required": true,
                        },
                        Object {
                          "message": "仅支持中英文字符、数字、空格、_、-、*、.、()",
                          "pattern": /\\^\\[a-zA-Z0-9_\\\\s\\\\-\\*\\.\\(\\)\\\\u4e00-\\\\u9fa5\\]\\+\\$/,
                        },
                        Object {
                          "message": "不可以为空",
                          "whitespace": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "validateTrigger": "onChange",
                  "valuePropName": "value",
                }
              }
              disabled={false}
              id="name"
              onChange={[Function]}
              onKeyDown={[Function]}
              type="text"
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-6 ant-form-item-label"
      >
        <label
          className="ant-form-item-required"
          htmlFor="ca"
          onClick={[Function]}
          title="证书内容 ( PEM格式 )"
        >
          证书内容 ( PEM格式 )
        </label>
      </div>
      <div
        className="ant-col ant-col-18 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <textarea
              className="ant-input"
              data-__field={
                Object {
                  "name": "ca",
                }
              }
              data-__meta={
                Object {
                  "initialValue": undefined,
                  "name": "ca",
                  "originalProps": Object {
                    "placeholder": "请将证书内容(包含证书链)复制粘贴到此处",
                    "rows": 10,
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "必填",
                      "required": true,
                    },
                    Object {
                      "message": "不可以为空",
                      "whitespace": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "必填",
                          "required": true,
                        },
                        Object {
                          "message": "不可以为空",
                          "whitespace": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "validateTrigger": "onChange",
                  "valuePropName": "value",
                }
              }
              id="ca"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="请将证书内容(包含证书链)复制粘贴到此处"
              rows={10}
              style={Object {}}
              value=""
            />
          </span>
        </div>
      </div>
    </div>
    <div
      className="ant-row ant-form-item"
    >
      <div
        className="ant-col ant-col-6 ant-form-item-label"
      >
        <label
          className="ant-form-item-required"
          htmlFor="pri"
          onClick={[Function]}
          title="证书私钥 ( PEM格式 )"
        >
          证书私钥 ( PEM格式 )
        </label>
      </div>
      <div
        className="ant-col ant-col-18 ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <textarea
              className="ant-input"
              data-__field={
                Object {
                  "name": "pri",
                }
              }
              data-__meta={
                Object {
                  "initialValue": undefined,
                  "name": "pri",
                  "originalProps": Object {
                    "placeholder": "请将证书私钥复制粘贴到此处",
                    "rows": 10,
                  },
                  "ref": null,
                  "rules": Array [
                    Object {
                      "message": "必填",
                      "required": true,
                    },
                    Object {
                      "message": "不可以为空",
                      "whitespace": true,
                    },
                  ],
                  "trigger": "onChange",
                  "validate": Array [
                    Object {
                      "rules": Array [
                        Object {
                          "message": "必填",
                          "required": true,
                        },
                        Object {
                          "message": "不可以为空",
                          "whitespace": true,
                        },
                      ],
                      "trigger": Array [
                        "onChange",
                      ],
                    },
                  ],
                  "validateTrigger": "onChange",
                  "valuePropName": "value",
                }
              }
              id="pri"
              onChange={[Function]}
              onKeyDown={[Function]}
              placeholder="请将证书私钥复制粘贴到此处"
              rows={10}
              style={Object {}}
              value=""
            />
          </span>
        </div>
      </div>
    </div>
  </div>
  <hr
    className="split-line sub-split-line"
  />
  <div
    className="footer-center"
  >
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    >
      <span>
        取消
      </span>
    </button>
    <button
      className="ant-btn ant-btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span>
        确定上传
      </span>
    </button>
  </div>
</form>
`;
